kpatch-build/kpatch-elf.c

```diff
@@ -328,9 +341,29 @@ void kpatch_create_symbol_list(struct kpatch_elf *kelf)
 					sym->name);
 
 			if (is_bundleable(sym)) {
-				if (sym->sym.st_value != 0)
-					ERROR("symbol %s at offset %lu within section %s, expected 0",
+				/*
+				 * function prologue generated by gcc6 on ppc64le has
+				 * the sequence:
+				 *
+				 *	.globl my_func
+				 *	.type my_func, @function
+				 *	.quad .TOC.-my_func
+				 * my_func:
+				 *	.reloc ., R_PPC64_ENTRY ; optional
+				 *	ld r2,-8(r12)
+				 *	add r2,r2,r12
+				 * 	.localentry my_func, .-my_func
+				 *
+				 * where .localentry for the functions, will generated
+				 * st_value of 0x8 for the local functions. Skip, this
+				 * case.
+				 */
+				if (sym->sym.st_value != 0) {
+					if (!gcc6_check() ||
+					   (sym->type != STT_FUNC || sym->sym.st_value != 0x8))
+						ERROR("symbol %s at offset %lu within section %s, expected 0",
 					      sym->name, sym->sym.st_value, sym->sec->name);
+				}
```

Comment:
With gcc 6 the local function symbol st_value is 0x8. Agree, I will re-work on a more reliable way for local entry detection and get rid of gcc6_check().